src/datepicker/datepicker.js

```diff
@@ -618,7 +618,16 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
         scope.$parent.$watch(getAttribute, function(value) {
           scope.watchData[key] = value;
           if (key === 'minDate' || key === 'maxDate') {
-            cache[key] = new Date(value);
+            var date = parseDate(value);
+            if(!date) {
+              var tryDate = new Date(value);
+              if(angular.isDate(tryDate) && !isNaN(tryDate)) {
```

Comment:
`undefined` is gets set.... this is to cover for an invalid attribute; example: a scope variable that doesn't exist.